src/sqlfluff/rules/L016.py

```diff
@@ -443,7 +443,7 @@ def _compute_source_length(cls, segments: Sequence[BaseSegment]) -> int:
             # source slice, and if we didn't correct for this, we'd count the
             # length of {{bi_ecommerce_orders}} roughly 10 times, resulting in
             # vast overcount of the source length.
-            if slice not in seen_slices:
+            if not segment.is_templated or slice not in seen_slices:
```

Comment:
See the pre-existing block of comments above to understand the intent of this code. It turns out that this code was accidentally ignoring **new** segments (introduced by earlier fixes). By definition, these new segments cannot be templated (because SQLFluff itself introduced them during lint/fix, i.e. after templating).

Reply:
Struggling to get my head round this, but isn't there a concern this will include any untemplated code that has been seen?

What's the precedence of `not` here?

Is this:

```py
if not (segment.is_templated or slice not in seen_slices):
```

or

```py
if (not segment.is_templated) or (slice not in seen_slices):
```

Reply:
The second one.

Reply:
I have another way to structure this. Warning: It's a bit like the "fixes" PR last week. 😛 

Just a sec...